src/test/java/org/broadinstitute/barclay/argparser/CommandLineArgumentParserTest.java

```diff
@@ -831,6 +831,41 @@ public void testOutOfRangesArguments(final String[] argv) throws Exception {
         clp.parseArguments(System.err, argv);
     }
 
+    @Test(expectedExceptions = CommandLineException.CommandLineParserInternalException.class)
+    public void testHiddenRequiredArgumentThrowException() throws Exception {
+        @CommandLineProgramProperties(summary = "tool with required and hidden argument",
+                oneLineSummary = "tool with required and hidden argument",
+                programGroup = TestProgramGroup.class)
+        final class ToolWithRequiredHiddenArgument {
+            // TODO: this does not blow up if it is a primitive, because it is not consider required
```

Comment:
Its a known issue, but results in somewhat surprising behavior. The parser treats any argument that has a non-null value as if it were "provided". Since primitives are never null, the parser is always satisfied that they've been set, which makes them appear to be optional even when they're required. The comment for the optional attribute in Argument describes this behavior.